lib/src/checks.dart

```diff
@@ -425,33 +437,49 @@ class CooldownCheck extends AbstractCheck {
 
   /// Get a key representing a [Context] depending on [type].
   int getKey(Context context) {
-    switch (type) {
-      case CooldownType.category:
-        if (context.guild != null) {
-          if ((context.channel as IGuildChannel).parentChannel != null) {
-            return (context.channel as IGuildChannel).parentChannel!.id.id;
-          }
-        }
-        return context.channel.id.id;
-      case CooldownType.channel:
-        return context.channel.id.id;
-      case CooldownType.command:
-        return context.command.hashCode;
-      case CooldownType.global:
-        return 0;
-      case CooldownType.guild:
-        return context.guild?.id.id ?? context.user.id.id;
-      case CooldownType.role:
-        if (context.member != null) {
-          if (context.member!.roles.isNotEmpty) {
-            return PermissionsUtils.getMemberHighestRole(context.member!).id.id;
-          }
-          return context.guild!.everyoneRole.id.id;
+    List<int> keys = [];
+
+    if (type.value & CooldownType.category.value != 0) {
+      if (context.guild != null) {
+        keys.add((context.channel as IGuildChannel).parentChannel?.id.id ?? context.channel.id.id);
+      } else {
+        keys.add(context.channel.id.id);
+      }
+    }
+
+    if (type.value & CooldownType.channel.value != 0) {
+      keys.add(context.channel.id.id);
+    }
+
+    if (type.value & CooldownType.command.value != 0) {
+      keys.add(context.command.hashCode);
+    }
+
+    if (type.value & CooldownType.global.value != 0) {
+      keys.add(0);
+    }
+
+    if (type.value & CooldownType.guild.value != 0) {
```

Comment:
I would write an util method for checking if bit value is true, there is so many duplicated code in there